### PR TITLE
Revive PR: Add Application Insights HealthCheck

### PR DESCRIPTION
**Pull request by cieciurm**
**What this PR does / why we need it**:
This PR revives #740 

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:
- Merged from `master`
- Code updated
- Tests updated
- GitHub actions updated

**Does this PR introduce a user-facing change?**:

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [x] Code compiles correctly
- [x] Created/updated tests
- [x] Unit tests passing
- [x] End-to-end tests passing
- [x] Extended the documentation
- [x] Provided sample for the feature

### CODE REVIEW

**Comment by cieciurm** [comment]
Hi @sungam3r,
It's ready for review.

File: src/HealthChecks.AzureApplicationInsights/HealthChecks.AzureApplicationInsights.csproj [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <PackageTags>HealthCheck;Azure;ApplicationInsights</PackageTags>
+    <Description>HealthChecks.AzureApplicationInsights is the health check package for Azure Application Insights.</Description>
+    <Version>$(HealthCheckAzureApplicationInsights)</Version>
```
Comment (sungam3r) [inline src/HealthChecks.AzureApplicationInsights/HealthChecks.AzureApplicationInsights.csproj:7]: undefined

File: build/versions.props [hunk 2]
```diff
@@ -9,6 +9,7 @@
     <HealthCheckAWSSqs>7.0.0</HealthCheckAWSSqs>
     <HealthCheckAWSSystemsManager>7.0.0</HealthCheckAWSSystemsManager>
     <HealthCheckAzureDigitalTwin>7.0.0</HealthCheckAzureDigitalTwin>
+    <HealthCheckAzureApplicationInsights>7.0.0</HealthCheckAzureApplicationInsights>
```
Comment (sungam3r) [inline build/versions.props:12]:
```suggestion
    <HealthCheckAzureApplicationInsights>7.0.0</HealthCheckAzureApplicationInsights>
    <HealthCheckAzureDigitalTwin>7.0.0</HealthCheckAzureDigitalTwin>
```

File: src/HealthChecks.AzureApplicationInsights/AzureApplicationInsightsHealthCheck.cs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        private readonly IHttpClientFactory _httpClientFactory;
+
+        /// <inheritdoc />
+        public AzureApplicationInsightsHealthCheck(string instrumentationKey, IHttpClientFactory httpClientFactory)
+        {
+            _instrumentationKey = Guard.ThrowIfNull(instrumentationKey, throwOnEmptyString: true);
+            _httpClientFactory = Guard.ThrowIfNull(httpClientFactory);
+        }
+
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
```
Comment (sungam3r) [inline src/HealthChecks.AzureApplicationInsights/AzureApplicationInsightsHealthCheck.cs:26]:
```suggestion
        /// <inheritdoc />
        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
```

File: src/HealthChecks.AzureApplicationInsights/AzureApplicationInsightsHealthCheck.cs [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        {
+            "https://dc.applicationinsights.azure.com",
+            "https://dc.applicationinsights.microsoft.com",
+            "https://dc.services.visualstudio.com"
+        };
+        private readonly string _instrumentationKey;
+
+        private readonly IHttpClientFactory _httpClientFactory;
+
+        /// <inheritdoc />
```
Comment (sungam3r) [inline src/HealthChecks.AzureApplicationInsights/AzureApplicationInsightsHealthCheck.cs:19]:
```suggestion
```

**Review by sungam3r — APPROVED** [review]

**Comment by sungam3r** [comment]
Please update root readme with new package.

**Comment by cieciurm** [comment]
Oops, my bad. It's been updated now - cc @sungam3r

File: README.md [hunk 5]
```diff
@@ -123,6 +124,7 @@ Install-Package AspNetCore.HealthChecks.Aws.Sqs
 Install-Package AspNetCore.HealthChecks.Aws.SystemsManager
 Install-Package AspNetCore.HealthChecks.Azure.IoTHub
 Install-Package AspNetCore.HealthChecks.AzureDigitalTwin
+Install-Package AspNetCore.HealthChecks.AzureApplicationInsights
```
Comment (sungam3r) [inline README.md:127]:
```suggestion
Install-Package AspNetCore.HealthChecks.AzureApplicationInsights
Install-Package AspNetCore.HealthChecks.AzureDigitalTwin
```

File: README.md [hunk 6]
```diff
@@ -64,52 +64,53 @@ HealthChecks repo provides following images:
...
+| Amazon SQS                 | [![Nuget](https://img.shields.io/nuget/dt/AspNetCore.HealthChecks.Aws.Sqs)](https://www.nuget.org/packages/AspNetCore.HealthChecks.Aws.Sqs)                         | [![Nuget](https://img.shields.io/nuget/v/AspNetCore.HealthChecks.Aws.Sqs)](https://www.nuget.org/packages/AspNetCore.HealthChecks.Aws.Sqs)
+| Amazon Systems Manager     | [![Nuget](https://img.shields.io/nuget/dt/AspNetCore.HealthChecks.Aws.SystemsManager)](https://www.nuget.org/packages/AspNetCore.HealthChecks.Aws.SystemsManager)   | [![Nuget](https://img.shields.io/nuget/v/AspNetCore.HealthChecks.Aws.SystemsManager)](https://www.nuget.org/packages/AspNetCore.HealthChecks.Aws.SystemsManager)
+| Azure Application Insights | [![Nuget](https://img.shields.io/nuget/dt/AspNetCore.HealthChecks.Azure.IoTHub)](https://www.nuget.org/packages/AspNetCore.HealthChecks.Azure.IoTHub)               | [![Nuget](https://img.shields.io/nuget/v/AspNetCore.HealthChecks.Azure.IoTHub)](https://www.nuget.org/packages/AspNetCore.HealthChecks.Azure.IoTHub)
```
Comment (sungam3r) [inline README.md:76]: wrong links to IoTHub

Comment (cieciurm) [reply README.md:76]: Ehh... My bad - sorry

**Comment by sungam3r** [comment]
I published preview version https://www.nuget.org/packages/AspNetCore.HealthChecks.AzureApplicationInsights/7.0.0-rc2.2.

Bad news - I accidentally published release version because of wrong tag in csproj, see https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/commit/60cedfba929163651ca26d74be2041efc0f68625.

**Pull request merged**